authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/role/RolePolicyProviderFactory.java

```diff
@@ -163,7 +164,7 @@ private void updateRoles(Policy policy, AuthorizationProvider authorization, Set
                         role = realm.getRoleById(roleName);
                     }
                 } else {
-                    ClientModel client = realm.getClientByClientId(clientId);
+                    ClientModel client = session.clients().getClientByClientId(realm, clientId);
```

Comment:
Are we going to replace any other occurrence of `RealmModel.getClientByClientId(String)` with `session.clients().getClientByClientId(RealmModel, String)`? It's not clear to me why we update it here but not on all other places. Are all other occurrences used becouse we don't have access to keycloak session there?

The same question applies to `RealmModel.getClientById(String)`.

Reply:
@vramik Good question. I guess I somehow assumed that I'll only change those occurrences which are relevant to this PR and the rest is/will be changed in a different PR. I'll revert these changes and it can be done in a separate task. WDYT?